trunk/source/Library-UltimateUtil/src/de/uni_freiburg/informatik/ultimate/util/CoreUtil.java

```diff
@@ -271,9 +271,6 @@ private static File writeFile(final String filename, final String[] content, fin
 
 	private static File writeFile(final String filename, final String content, final boolean append)
 			throws IOException {
-		if (content == null || content.isEmpty()) {
-			return null;
-		}

```

Comment:
:question:

Reply:
It is possible that we create empty YAML witnesses (e.g. if there are no loops), so we should still create the file. I checked the usage of the method, it is only used for witnesses and some stat outputs.

Reply:
:+1: I was a bit surprised there was even a check like this in the first place -- I'd say a condition like this should be up to the caller

Reply:
I agree for content.isEmpty(), but for content == null, this will crash